cmd/api/src/database/migration/migrations/v5.12.0.sql

```diff
@@ -29,4 +29,10 @@ VALUES
 ON CONFLICT DO NOTHING;
 
 ALTER TABLE user_sessions
-    ADD COLUMN IF NOT EXISTS flags jsonb;
\ No newline at end of file
+    ADD COLUMN IF NOT EXISTS flags jsonb;
+
+ALTER TABLE asset_groups
```

Comment:
We want to ensure these migrations are idempotent. 

There's not an `IF NOT EXISTS` option for `ADD CONSTRAINT` so you have to drop it with `DROP CONSTRAINT IF EXISTS` and then add it.

Reply:
This brings up another potential issue with running the migration script: handling existing null or duplicate entries to maintain idempotence. The script could fail when adding constraints if it encounters null or duplicated entries in the existing database.

Referencing the original Jira ticket about duplicate names/tags, which suggests appending a number to duplicate asset group tags: Should our script preemptively address existing null or duplicate entries before executing this migration? If the ALTER TABLE statement encounters NULL values in the name or tag column, it will fail.

I propose the following approach to ensure robustness:

1. Clean up any NULL values by assigning them default values.
2. Verify the absence of NULLs before enforcing the NOT NULL constraint.
3. Apply the NOT NULL constraint only if no NULL values are found.

Additionally, I believe we should integrate logic to rename existing duplicates. Your thoughts on this approach and any additional considerations?